src/modules/75-cf/pages/onboarding/__test__/CreateAFlagView.test.tsx

```diff
@@ -0,0 +1,147 @@
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]')
+    expect(selectInput).toBeVisible()
+    expect(selectInput).toHaveValue('')
+    expect(selectInput).toHaveAttribute('placeholder', 'cf.onboarding.selectOrCreateFlag')
+    expect(screen.queryByTestId('ffOnboardingSelectedFlag')).not.toBeInTheDocument()
+    expect(fetchFlags).toHaveBeenCalled()
+
+    userEvent.click(selectInput as TargetElement)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+  })
+
+  test('It should filter Select options by typed text', async () => {
+    renderComponent()
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]') as TargetElement
+    expect(document.querySelector('input[id="selectOrCreateFlag"]')).toBeVisible()
+    expect(fetchFlags).toHaveBeenCalled()
+
+    userEvent.click(selectInput)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+    expect(dropdownOptions[1]).toHaveTextContent('XYZ Flag')
+    expect(dropdownOptions[2]).toHaveTextContent('ABC Flag')
+
+    userEvent.type(selectInput, 'hello')
+    expect(fetchFlags).toBeCalled()
+    expect(dropdownOptions).toHaveLength(1)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+  })
+
+  test('It should set the selected dropdown option as the selected flag for Onboarding', () => {
+    renderComponent()
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]')
+    expect(selectInput).toBeVisible()
+    expect(screen.queryByTestId('ffOnboardingSelectedFlag')).not.toBeInTheDocument()
+    expect(fetchFlags).toHaveBeenCalled()
+
+    userEvent.click(selectInput as TargetElement)
+
+    const dropdownOptions = document.getElementsByTagName('li')
+    expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE)
+    expect(dropdownOptions[0]).toHaveTextContent('hello world')
+
+    userEvent.click(dropdownOptions[0])
+
+    expect(selectInput).toHaveValue('hello world')
+    expect(setSelectedFlag).toBeCalledWith(mockFeatureFlags.features[0])
+  })
+
+  test('It should display a label for the selected flag name and id', async () => {
+    renderComponent({ selectedFlag: mockFeatureFlags.features[0] as any })
+
+    const selectInput = document.querySelector('input[id="selectOrCreateFlag"]') as TargetElement
+    expect(document.querySelector('input[id="selectOrCreateFlag"]')).toBeVisible()
+    expect(fetchFlags).toBeCalled()
+    expect(selectInput).toHaveValue('hello world')
+
+    expect(screen.getByTestId('ffOnboardingSelectedFlag')).toBeVisible()
+    expect(screen.getByTestId('ffOnboardingSelectedFlag').textContent).toMatch(
+      /cf.onboarding.youreUsinghello world.+hello_world.+/g
+    )
```

Comment:
Not a problem, but you can use `toHaveTextContent` to make it read _slightly_ better
```suggestion
    expect(screen.getByTestId('ffOnboardingSelectedFlag').toHaveTextContent(
      /cf\.onboarding\.youreUsinghello world.+hello_world/
    )
```

Reply:
Didn't know it accepted a RegExp

Reply:
It doesn't accept regexp...

Reply:
It does indeed https://github.com/testing-library/jest-dom#tohavetextcontent

![Screenshot 2022-09-27 at 11 05 13](https://user-images.githubusercontent.com/7167253/192498447-e1ed0d89-2287-4901-87ca-ae32c61751eb.png)